src/util/index.ts

```diff
@@ -0,0 +1,76 @@
+import type { APIEmbed } from 'discord.js';
+
+/**
+ * Truncate a text to a provided length using a provided splitcharacter
+ *
+ * @param text - Text to truncate
+ * @param len - Length to truncate to
+ * @param splitChar - Split character to use
+ * @returns The truncated text
+ */
+export function truncate(text: string, len: number, splitChar = ' '): string {
+	if (text.length <= len) return text;
+	const words = text.split(splitChar);
+	const res: string[] = [];
+	for (const word of words) {
+		const full = res.join(splitChar);
+		if (full.length + word.length + 1 <= len - 3) {
+			res.push(word);
+		}
+	}
+
+	const resText = res.join(splitChar);
+	return resText.length === text.length ? resText : `${resText.trim()}...`;
+}
+
+export const LIMIT_EMBED_DESCRIPTION = 4_048 as const;
+export const LIMIT_EMBED_TITLE = 256 as const;
+export const LIMIT_EMBED_FIELDS = 25 as const;
+export const LIMIT_EMBED_FIELD_NAME = 256 as const;
+export const LIMIT_EMBED_FIELD_VALUE = 1_024 as const;
+export const LIMIT_EMBED_AUTHOR_NAME = 256 as const;
+export const LIMIT_EMBED_FOOTER_TEXT = 2_048 as const;
```

Comment:
This shouldn't be on consts too?

Reply:
it's contextual where it's used, which makes sense imo